open-api-spec.yaml

```diff
@@ -340,127 +324,122 @@ paths:
           content:
             application/json:
               schema:
-                type: array
-                items:
-                  $ref: '#/components/schemas/list-outgoing-payments'
+                type: object
+                properties:
+                  pagination:
```

Comment:
I've dropped the pagination query parameter changes from this pr to contain it pagination response.
I'll open a second pr that implements :point_up_2:

Reply:
I feel like wherever you look, that is done differently.  

I've seen `count` instead of `first`/`last` but like I said before, that can be derived from the list of results anyways, so it's not really necessary. 

I do think, however, that  `hasPreviousPage`/`hasNextPage` add value so I would vote to keep those. 

I don't have strong feelings for `startCursor` and `endCursor` because they, also, can be derived from the results. It's just a bit more complicated.

I don't like `<resource-name>s` instead of `result` because responses have to be treated differently. But maybe that is not a problem :thinking:

Reply:
`count` would only exist as a query parameter.
So, the query params would change from `first`/`last` + `cursor` to `after`/`before` + `count`

I've seen things (including the linked post) recommending opaque cursors, so I'm leaning toward not having the client derive them from the results.
Maybe the `pageInfo`/`pagination` in the response could be:
```
  cursor: {
    after: <opaque-cursor>
    hasNextPage: true
  }
  ...
```
```
  cursor: {
    before: <opaque-cursor>
    hasPreviousPage: true
  }
  ...
```
I don't have a strong opinion on what we call the field that includes the list.

Reply:
> So, the query params would change from `first`/`last` + `cursor` to `after`/`before` + `count`

Yes, I've seen this combination. I'm happy either way. The changes shouldn't take too long. 

I like the `cursor` objects. I also don't have strong feelings about the name of the field with the list.

Reply:
I've opened an issue to handle it in a separate pr:
https://github.com/interledger/open-payments/issues/136